prefect_aws/s3.py

```diff
@@ -147,6 +147,163 @@ async def example_s3_upload_flow():
     return key
 
 
+@task
+async def s3_copy(
+    source_path: str,
+    target_path: str,
+    source_bucket_name: str,
+    aws_credentials: AwsCredentials,
+    target_bucket_name: Optional[str] = None,
+    aws_client_parameters: AwsClientParameters = AwsClientParameters(),
+    **copy_kwargs,
+) -> str:
+    """Uses S3's internal
+    [CopyObject](https://docs.aws.amazon.com/AmazonS3/latest/API/API_CopyObject.html)
+    to copy objects within or between buckets. To copy objects between buckets, the
+    credentials must have permission to read the source object and write to the target
+    object.
+
+    Args:
+        source_path: The path to the object to copy. Can be a string or `Path`.
+        target_path: The path to copy the object to. Can be a string or `Path`.
+        source_bucket_name: The bucket to copy the object from.
+        aws_credentials: Credentials to use for authentication with AWS.
+        target_bucket_name: The bucket to copy the object to. If not provided, defaults
+            to `source_bucket`.
+        aws_client_parameters: Custom parameter for the boto3 client initialization.
+        **copy_kwargs: Additional keyword arguments to pass to `S3Client.copy_object`.
+
+    Returns:
+        The path that the object was copied to. Excludes the bucket name.
+
+    Examples:
+
+        Copy notes.txt from s3://my-bucket/my_folder/notes.txt to
+        s3://my-bucket/my_folder/notes_copy.txt.
+
+        ```python
+        from prefect import flow
+        from prefect_aws import AwsCredentials
+        from prefect_aws.s3 import s3_copy
+
+        aws_credentials = AwsCredentials.load("my-creds")
+
+        @flow
+        async def example_copy_flow():
+            await s3_copy(
+                source_path="my_folder/notes.txt",
+                target_path="my_folder/notes_copy.txt",
+                source_bucket_name="my-bucket",
+                aws_credentials=aws_credentials,
+            )
+
+        example_copy_flow()
+        ```
+
+        Copy notes.txt from s3://my-bucket/my_folder/notes.txt to
+        s3://other-bucket/notes_copy.txt.
+
+        ```python
+        from prefect import flow
+        from prefect_aws import AwsCredentials
+        from prefect_aws.s3 import s3_copy
+
+        aws_credentials = AwsCredentials.load("shared-creds")
+
+        @flow
+        async def example_copy_flow():
+            await s3_copy(
+                source_path="my_folder/notes.txt",
+                target_path="notes_copy.txt",
+                source_bucket_name="my-bucket",
+                aws_credentials=aws_credentials,
+                target_bucket_name="other-bucket",
+            )
+
+        example_copy_flow()
+        ```
+
+    """
+    logger = get_run_logger()
+
+    s3_client = aws_credentials.get_boto3_session().client(
+        "s3", **aws_client_parameters.get_params_override()
+    )
```

Comment:
I'd recommend using `get_client` to ensure that the client parameters on `aws_credentials` are respected.

```suggestion
    s3_client = aws_credentials.get_client("s3")
```